pkg/util/gvkr.go

```diff
@@ -0,0 +1,91 @@
+package util
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	defaultVersion  = "v1"
+	defaultGroup    = "apps"
+	defaultKind     = "Deployment"
+	defaultResource = "deployments"
+)
+
+// GroupVersionKindResource provides unified structure for schema.GroupVersionKind and Resource
+type GroupVersionKindResource struct {
+	Group    string `json:"group"`
+	Version  string `json:"version"`
+	Kind     string `json:"kind"`
+	Resource string `json:"resource"`
+}
+
+func (gvkr GroupVersionKindResource) GroupVersionKind() schema.GroupVersionKind {
+	return schema.GroupVersionKind{Group: gvkr.Group, Version: gvkr.Version, Kind: gvkr.Kind}
+}
+
+func (gvkr GroupVersionKindResource) GroupVersion() schema.GroupVersion {
+	return schema.GroupVersion{Group: gvkr.Group, Version: gvkr.Version}
+}
+
+func (gvkr GroupVersionKindResource) GroupResource() schema.GroupResource {
+	return schema.GroupResource{Group: gvkr.Group, Resource: gvkr.Resource}
+}
+
+func (gvkr GroupVersionKindResource) GVKString() string {
+	return gvkr.Group + "/" + gvkr.Version + "." + gvkr.Kind
+}
+
+// ParseGVKR returns GroupVersionKindResource for specified apiVersion (groupVersion) and Kind
+func ParseGVKR(restMapper meta.RESTMapper, apiVersion string, kind string) (GroupVersionKindResource, error) {
+	var group, version, resource string
+
+	// if apiVersion is not specified, we suppose the default one should be used
+	if apiVersion == "" {
+		group = defaultGroup
+		version = defaultVersion
+	} else {
+		groupVersion, err := schema.ParseGroupVersion(apiVersion)
+		if err != nil {
+			return GroupVersionKindResource{}, err
+		}
+
+		group = groupVersion.Group
+		version = groupVersion.Version
+	}
+
+	// if kind is not specified, we suppose that default one should be used
+	if kind == "" {
+		kind = defaultKind
+		resource = defaultResource
```

Comment:
Looks like `resource` initialization was refactored into `getResrouce` below, you can remove this, right?

Reply:
Good catch! thx